src/main/java/io/jenkins/plugins/pipelinegraphview/treescanner/PipelineNodeTreeScanner.java

```diff
@@ -352,12 +377,15 @@ private void buildGraph() {
          * Returns the origin of any unhandled exception for this node, or null if none
          * found.
          */
-        private @CheckForNull BlockEndNode<?> getUnhandledException(@NonNull FlowNode node) {
+        private @CheckForNull BlockEndNode<?> getUnhandledException(@NonNull FlowNode node, boolean hasStage) {
             // Check for an unhandled exception.
             ErrorAction errorAction = node.getAction(ErrorAction.class);
-            // If this is a Jenkins failure exception, then we don't need to add a new node
-            // - it will come from an existing step.
-            if (errorAction != null && !PipelineNodeUtil.isJenkinsFailureException(errorAction.getError())) {
+            if (errorAction != null) {
+                // If this is a Jenkins failure exception, then we don't need to add a new node
+                // - it will come from an existing step if there is a stage for the step to be part of.
+                if (hasStage && PipelineNodeUtil.isJenkinsFailureException(errorAction.getError())) {
```

Comment:
I think this will fix it for no stages, but does this fix `withCredentials` in a stage? I think that's where I got stuck because its a block scoped step, it wasn't handled.